unison-cli/src/Unison/Codebase/Editor/Output.hs

```diff
@@ -323,7 +323,8 @@ isFailure o = case o of
   BadMainFunction {} -> True
   CreatedNewBranch {} -> False
   BranchAlreadyExists {} -> True
-  FindNoLocalMatches {} -> True
+  -- we do a global search after finding no local matches, so let's not call this a failure yet
```

Comment:
Unrelated to this PR but this whole function is weird, and this change highlights that. I think we should get rid of it and stop classifying stanzas in transcripts as "I expect success" or "I expect an error" - couldn't the golden test output just be inspected, and committed if it looks right, without switching on this bool?

Reply:
I think the idea was to have a sense of overall success or failure for a transcript; e.g. we currently run `transcripts` and count the number of successes and failures, and list the failures, so you can go back and check. Maybe that's misguided today though.

Reply:
Maybe "transcript failed" or "transcript succeeded" also doesn't mean much locally, maybe we'd need to use `git` to decide if there was a "failure", and maybe that's a good idea.